QgisModelBaker/gui/panel/dataset_combobox.py

```diff
@@ -0,0 +1,166 @@
+)
+
+from QgisModelBaker.utils.qt_utils import slugify
+from QgisModelBaker.libili2db.ili2dbconfig import Ili2DbCommandConfiguration
+from QgisModelBaker.libili2db.globals import DbIliMode
+from ...libqgsprojectgen.db_factory.db_simple_factory import DbSimpleFactory
+class DatasetSourceModel(QStandardItemModel):
+    class Roles(Enum):
+        DATASETNAME = Qt.UserRole + 1
+        MODEL_TOPIC = Qt.UserRole + 2
+        BASKET_TID = Qt.UserRole + 3
+        # The SCHEMA_TOPIC_IDENTIFICATOR is a combination of db parameters and the topic
+        # This because a dataset is usually valid per topic and db schema
+        SCHEMA_TOPIC_IDENTIFICATOR = Qt.UserRole + 4
+
+        def __int__(self):
+            return self.value
+
+    def __init__(self):
+        super().__init__()
+        self.schema_baskets = {}
+
+    def refresh(self):
+        self.beginResetModel()
+        self.clear()
+        for schema_identificator in self.schema_baskets.keys():
+            for basket in self.schema_baskets[schema_identificator]:
+                item = QStandardItem()
+                item.setData(basket['datasetname'], int(Qt.DisplayRole))
+                item.setData(basket['datasetname'], int(DatasetSourceModel.Roles.DATASETNAME))
+                item.setData(basket['topic'], int(DatasetSourceModel.Roles.MODEL_TOPIC))
+                item.setData(basket['basket_t_id'], int(DatasetSourceModel.Roles.BASKET_TID))
+                item.setData(f"{schema_identificator}_{slugify(basket['topic'])}", int(DatasetSourceModel.Roles.SCHEMA_TOPIC_IDENTIFICATOR))
+                self.appendRow(item)
+        self.endResetModel()
+
+    def reload_schema_baskets(self, db_connector, schema_identificator):
+        baskets_info = db_connector.get_baskets_info()
+        baskets = []
+        for record in baskets_info:
+            basket = {}
+            basket['datasetname'] = record['datasetname']
+            basket['topic'] = record['topic']
+            basket['basket_t_id'] = record['basket_t_id']
+            baskets.append(basket)
+        self.schema_baskets[schema_identificator] = baskets
+        self.refresh()
+
+    def schema_baskets_loaded(self, schema_identificator):
+        return schema_identificator in self.schema_baskets
+
+class DatasetCombobox(QComboBox):
+    def __init__(self, parent=None):
+        QWidget.__init__(self, parent)
+        self.setSizeAdjustPolicy(QComboBox.AdjustToContents)
+
+        self.db_simple_factory = DbSimpleFactory()
+        self.basket_model = DatasetSourceModel()
+        self.filtered_model = QSortFilterProxyModel()
+        self.filtered_model.setSourceModel(self.basket_model)
+        self.filtered_model.setFilterRole(int(DatasetSourceModel.Roles.SCHEMA_TOPIC_IDENTIFICATOR))
+        self.setModel(self.filtered_model)
+        self.setEnabled(False)
+
+    def set_current_layer(self, layer):
+        if self.isEnabled():
+            self.currentIndexChanged.disconnect(self.store_basket_tid)
```

Comment:
Will we ever need this combobox for another action than selecting the currently active dataset?
If yes it would be good to separate the code to create a (read only) combobox and the code to update the project variables.

Reply:
You are thinking about the selection of the dataset on data import?
I thought about that, but this combobox diverts from it there. Though we select the datasets here it's actually a selection of the basket from the available baskets for the topic of the current layer. So I don't see much synergies I could use here.

Reply:
Ok, let's put that away and refactor if we have real requirements.